packages/amplify-category-notifications/src/channel-sms.ts

```diff
@@ -55,32 +54,28 @@ export const enable = async (context:$TSContext):Promise<$TSAny> => {
   };
 
   spinner.start('Enabling SMS channel.');
-  return new Promise((resolve, reject) => {
-    context.exeInfo.pinpointClient.updateSmsChannel(params, (err:$TSAny, data:$TSAny) => {
-      if (err) {
-        if (!isAmplifyCLIPulling(context)) {
-          spinner.fail('enable channel error');
-        }
-        const enableChannelErrorResponse = buildPinpointChannelResponseError(ChannelAction.ENABLE, deploymentType,
-          channelName, err);
-        reject(enableChannelErrorResponse);
-        return;
-      }
-      spinner.succeed(`The ${channelName} channel has been successfully enabled.`);
-      context.exeInfo.serviceMeta.output[channelName] = data.SMSChannelResponse;
-      const enableChannelResponse = buildPinpointChannelResponseSuccess(ChannelAction.ENABLE, deploymentType,
-        channelName, data.SMSChannelResponse);
-      resolve(enableChannelResponse);
+
+  try {
+    const data = await context.exeInfo.pinpointClient.updateSmsChannel(params).promise();
+    context.exeInfo.serviceMeta.output[channelName] = data.SMSChannelResponse;
+    spinner.succeed(`The ${channelName} channel has been successfully enabled.`);
+
+    return buildPinpointChannelResponseSuccess(ChannelAction.ENABLE, deploymentType, channelName, data.SMSChannelResponse);
+  } catch (e) {
+    spinner.stop();
+    throw amplifyFaultWithTroubleshootingLink('NotificationsChannelEmailFault', {
```

Comment:
we should pass the underlying stack into the amplfiy fault as well (applies to lots of places). Not sure how feasible it is to update all of them